src/multicore.rs

```diff
@@ -32,17 +32,15 @@ where
         .build()
         .expect("If the platform can spawn threads, then this call will work.");
 
-    let mut iter_rng = rng.clone();
-    let iter = match SieveIterator::new(&mut iter_rng, sieve_factory)? {
+    let iter = match SieveIterator::new(rng, sieve_factory)? {
         Some(iter) => iter,
         None => return Ok(None),
     };
 
     threadpool.install(|| {
-        Ok(iter.par_bridge().find_any(|c| {
-            let mut rng = rng.clone();
-            predicate(&mut rng, c)
-        }))
+        Ok(iter
+            .par_bridge()
+            .find_map_any(|(mut rng, c)| if predicate(&mut rng, &c) { Some(c) } else { None }))
```

Comment:
Hm, wait, how does that work? Why don't we need to clone the RNG anymore?

Reply:
Clone on an RNG is not a really good idea as either forks of the clone will have a similar internal state and will yield the same bits.

Hence the idea to use [`SeedableRng::fork`](https://github.com/rust-random/rand_core/pull/17) instead (which needs to take a `&mut` to mutate the internal state of the source RNG).

rand_chacha used to carry `Clone`: https://docs.rs/rand_chacha/latest/rand_chacha/struct.ChaCha20Core.html#impl-Clone-for-ChaCha20Core (rand_chacha is getting deprecated)
chacha20:rngs does not anymore: https://docs.rs/chacha20/0.10.0-rc.5/chacha20/struct.ChaCha20Rng.html

Here instead, the rng is injected by the iterator in each of the items (using fork).
It's a couple of lines down from this change.